reproducibility_project/src/engines/mcccs/project.py

```diff
@@ -165,59 +246,130 @@ def cool_finished(job):
 @Project.label
 def equil_finished(job):
     """Check if equil stage is finished."""
-    step = "equil"
+    try:
+
+        step = "equil" + str(job.doc.equil_replicates_done - 1)
+    except (KeyError, AttributeError):
+        step = "equil" + "0"
+
     run_file = job.ws + "/run.{}".format(step)
     if job.isfile("run.{}".format(step)):
         with open(run_file) as myfile:
             if "Program ended" in myfile.read():
                 return True
             else:
                 return False
+    else:
+        return False
+
+
+def sanitize_npt_log(step):
+    """Sanitize the output logs for NPT simulations."""
+    import numpy as np
+
+    files = glob("fort*12*{}*".format(step))
+    arrays = []
+    for filecurrent in files:
+        array = np.genfromtxt(filecurrent, skip_header=1)
+        arrays.append(array)
+    arrays = np.vstack(arrays)
+    arrays[:, 0] = arrays[:, 0] / 10
+    arrays[:, 1] = arrays[:, 1] / 10
+    arrays[:, 2] = arrays[:, 2] / 10
+    arrays[:, 3] = arrays[:, 3] * 0.008314410016255453
+    arrays[:, 4] = arrays[:, 4]
+    return arrays
+
+
+def system_equilibrated(job):
+    """Check if the system is equilibrated."""
+    orig_dir = os.getcwd()
+    os.chdir(job.ws)
+    files = glob("fort*12*{}*".format("equil"))
+    if len(files) < 2:  # at least do two loops of equilibration
+        print("equils done is less than 2")
+        return False
+    if job.sp.ensemble == "NPT":
+        equil_log = sanitize_npt_log("equil")
+
+    # run pymbar
+    return True
```

Comment:
You dont need to use os.chdir since you have the job.

you can do it in multiple ways

Refer to: https://docs.signac.io/projects/flow/en/latest/api.html#flow.with_job

```suggestion

def system_equilibrated(job):
    """Check if the system is equilibrated."""
    with job:
        files = glob("fort*12*{}*".format("equil"))
        if len(files) < 2:  # at least do two loops of equilibration
            print("equils done is less than 2")
            return False
        if job.sp.ensemble == "NPT":
            equil_log = sanitize_npt_log("equil")

    # run pymbar
    return True
```

or

```suggestion
@flow.with_job
def system_equilibrated(job):
    """Check if the system is equilibrated."""
    files = glob("fort*12*{}*".format("equil"))
    if len(files) < 2:  # at least do two loops of equilibration
        print("equils done is less than 2")
        return False
    if job.sp.ensemble == "NPT":
        equil_log = sanitize_npt_log("equil")

    # run pymbar
    return True
```